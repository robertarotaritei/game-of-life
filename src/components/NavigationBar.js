import React from 'react';
import UserStore from '../stores/UserStore';
import { runInAction } from 'mobx';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Container from '@material-ui/core/Container';
import { Redirect } from 'react-router-dom';

class NavigationBar extends React.Component {
    constructor() {
        super();

        this.state = {
            redirectLogin: false
        }
        runInAction(() => {
            UserStore.key = sessionStorage.getItem('key');
        });
    }

    doLogout() {
        sessionStorage.removeItem('key');
        sessionStorage.removeItem('username');

        runInAction(() => {
            UserStore.key = '';
            UserStore.username = '';
        })
    }

    doLogin = () => {
        this.setState({
            redirectLogin: true
        })
    }

    renderRedirect = () => {
        if (this.state.redirectLogin) {
            return <Redirect to='/login' />
        }
    }

    render() {
        return (
            <AppBar className='navbar' position="fixed" style={{ background: '#063A4C' }}>
                <Toolbar variant="dense" style={{ maxHeight: '2rem' }}>
                <Container disableGutters>
                    <p style={{ color: '#9BE8FF', marginLeft: '6em' }} >
                        Welcome {UserStore.username}
                    </p>
                    </Container>
                    <Container disableGutters>
                        <div style={{ float: 'right' }}>
                            {this.renderRedirect()}
                            <button className='btnNav' onClick={this.props.renderDashboard} style={{ backgroundColor: this.props.selectedPage === 'dashboard' ? '#009ECE' : null }}>
                                Dashboard
                            <span className="btnNav__glitch"></span>
                            </button>
                            <button className='btnNav' onClick={this.props.renderGameHistory} style={{ backgroundColor: this.props.selectedPage === 'history' ? '#009ECE' : null }} >
                                Game History
                            <span className="btnNav__glitch"></span>
                            </button>
                            <button className='btnNav' onClick={this.props.renderWelcome} style={{ backgroundColor: this.props.selectedPage === 'about' ? '#009ECE' : null }} >
                                About
                            <span className="btnNav__glitch"></span>
                            </button>
                            {this.props.loggedIn ? (
                                <button className='btnNav' data-testid={'logOut'} onClick={this.doLogout} style={{ marginLeft: '1.5em', marginRight: '6em' }}>
                                    Log out
                                    <span className="btnNav__glitch"></span>
                                </button>
                            ) : (
                                    <button className='btnNav' data-testid={'logIn'} onClick={this.doLogin} style={{ marginLeft: '1.5em', marginRight: '6em' }}>
                                        Log in
                                        <span className="btnNav__glitch"></span>
                                    </button>
                                )}
                        </div>
                    </Container>
                </Toolbar>
            </AppBar>
        );
    }
}

export default NavigationBar;