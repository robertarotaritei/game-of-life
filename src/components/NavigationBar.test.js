import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NavigationBar from './NavigationBar';
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";

test("NavigationBar renders without crashing", () => {
  shallow(<NavigationBar />);
});

it('NavigationBar renders correctly', () => {
  const tree = renderer
    .create(<NavigationBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render NavigationBar Log in Button', () => {
  const { getByText } = render(<NavigationBar />);
  const linkElement = getByText(/Log in/i);
  expect(linkElement).toBeInTheDocument();
});
