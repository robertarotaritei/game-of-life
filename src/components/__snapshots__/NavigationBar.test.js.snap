// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary navbar mui-fixed MuiPaper-elevation4"
  style={
    Object {
      "background": "#063A4C",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
    style={
      Object {
        "maxHeight": "2rem",
      }
    }
  >
    <div
      className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
    >
      <p
        style={
          Object {
            "color": "#9BE8FF",
            "marginLeft": "6em",
          }
        }
      >
        Welcome 
        
      </p>
    </div>
    <div
      className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
    >
      <div
        style={
          Object {
            "float": "right",
          }
        }
      >
        <button
          className="btnNav"
          style={
            Object {
              "backgroundColor": null,
            }
          }
        >
          Dashboard
          <span
            className="btnNav__glitch"
          />
        </button>
        <button
          className="btnNav"
          style={
            Object {
              "backgroundColor": null,
            }
          }
        >
          Game History
          <span
            className="btnNav__glitch"
          />
        </button>
        <button
          className="btnNav"
          style={
            Object {
              "backgroundColor": null,
            }
          }
        >
          About
          <span
            className="btnNav__glitch"
          />
        </button>
        <button
          className="btnNav"
          data-testid="logIn"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "1.5em",
              "marginRight": "6em",
            }
          }
        >
          Log in
          <span
            className="btnNav__glitch"
          />
        </button>
      </div>
    </div>
  </div>
</header>
`;
